profile/src/main/java/com/scottlogic/deg/profile/reader/validation/ConstraintTypeValidator.java

```diff
@@ -0,0 +1,88 @@
+package com.scottlogic.deg.profile.reader.validation;
+
+import com.scottlogic.deg.common.ValidationException;
+import com.scottlogic.deg.common.profile.DateTimeGranularity;
+import com.scottlogic.deg.common.profile.Field;
+import com.scottlogic.deg.common.profile.FieldType;
+import com.scottlogic.deg.common.profile.NumericGranularity;
+import com.scottlogic.deg.common.util.Defaults;
+import com.scottlogic.deg.common.util.NumberUtils;
+import com.scottlogic.deg.generator.fieldspecs.whitelist.DistributedList;
+import com.scottlogic.deg.profile.common.ConstraintType;
+import com.scottlogic.deg.profile.reader.InvalidProfileException;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import static com.scottlogic.deg.common.profile.FieldType.*;
+import static com.scottlogic.deg.profile.common.ConstraintType.GRANULAR_TO;
+import static com.scottlogic.deg.profile.common.ConstraintType.IS_NULL;
+
+public class ConstraintTypeValidator {
+
+    public static void validateTypeIs(Field field, ConstraintType type, FieldType s) {
+        if (field.getType() == null){
+            throw new ValidationException("Field [" + field.name + "]: is not typed; add its type to the field definition");
+        }
+        if (field.getType() != s){
+            throw new ValidationException("Field [" + field.name + "]: is type " + field.getType() + " , but you are trying to apply a " + type.propertyName + " constraint which requires " + s);
+        }
+    }
+
+    public static void validateAny(Field field, ConstraintType type, Object value) {
```

Comment:
I'm not sure this method name captures what the method does (a.k.a. I don't understand what's going on here)